### ISSUE TITLE
aws_ram_resource_share Data Source?

### ISSUE THREAD

**Issue by MMarulla** [terraform-providers/terraform-provider-aws#8280]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

In my scenario, I have a Transit Gateway running in a master account and wish to attach VPCs running in other (child) accounts within the same organization.  I can use the aws_ec2_transit_gateway data source when actually attaching to the Transit Gateway, but I first have to share the Transit Gateway with the child account.  The logical way to do this would be to place a aws_ram_principal_association resource in the child account's Terraform script, using the provider of the master account, referencing the child account's account number, and the ARN of the Resource Share.  Placing it in the master account's Terraform would be more difficult since that account does not have access to the account numbers of the child accounts.  The issue is with getting the Resource Share ARN - It seems there is currently no Data Source for the aws_ram_resource_share resource.

### New or Affected Resource(s)

* aws_ram_resource_share (data source)

### Potential Terraform Configuration

```hcl
data "aws_ram_resource_share" "um-ss-tgw-share" {
  name = "UM Shared Services GTW Share"
}

resource "aws_ram_principal_association" "um-ss-tgw" {
  provider = "aws.master"
  principal = "${var.target_account}"
  resource_share_arn = "${data.aws_ram_resource_share.um-ss-tgw-share.arn}"
}
```

**Comment by bflad** [comment]
A new `aws_ram_resource_share` data source has been merged and will release with version 2.11.0 of the Terraform AWS Provider, likely later this week. 👍

**Comment by bflad** [comment]
This has been released in [version 2.11.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2110-may-17-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
